cmd/object-handlers.go

```diff
@@ -736,7 +736,7 @@ func (api objectAPIHandlers) CopyObjectHandler(w http.ResponseWriter, r *http.Re
 
 	// Deny if WORM is enabled. If operation is key rotation of SSE-S3 encrypted object
 	// allow the operation
-	if globalWORMEnabled && !(cpSrcDstSame && crypto.S3.IsRequested(r.Header)) {
+	if globalWORMEnabled && !(cpSrcDstSame && crypto.SSEC.IsRequested(r.Header) && crypto.SSECopy.IsRequested(r.Header)) {
```

Comment:
this check was only for SSE-S3 in WORM mode to allow rotating vault keys. For SSE-C key-rotation shouldn't be allowed, correct @aead ?

Reply:
I'm actually not sure about that. Actually, performing a key rotation does not change the object. Furthermore, this may be required if a SSE-C key gets compromised. On the other side, allowing key rotation may lead to the situation that an object becomes unreadable after upload (e.g. loss of SSE-C key)

Reply:
Update: We decided to not allow SSE-C key rotation in WORM mode. Will remove that code.